crates/sui-core/src/authority_client.rs

```diff
@@ -115,67 +119,135 @@ impl AuthorityAPI for NetworkAuthorityClient {
         &self,
         transaction: Transaction,
     ) -> Result<TransactionInfoResponse, SuiError> {
-        self.client()
+        self.metrics
+            .handle_transaction_request_bytes
+            .observe(mem::size_of_val(&transaction) as f64);
```

Comment:
I am not sure this is a reliable indication of the size of the request / response. My understanding is that `size_of_val` returns the size of the structure, but does not follow pointers, so to say. So for example for a Vec it will return that the size of 2 * 8 (64 bites) which are the size + pointer to header allocation. This is catastrophic as the data we care about are in structures of variable length on the heap pointed to by pointers. At least this is my suspicion -- can we confirm?

Reply:
I just confirmed this will not work sadly. This program:
```
use std::mem;

fn main() {

    let store : Vec<u64> = (0..10000).into_iter().collect();
    println!("Vec len: {}", store.len());
    
    let size = mem::size_of_val(&store);
    println!("Size of val: {}", size);
}
```
Prints
```
Vec len: 10000
Size of val: 24
```

Reply:
Maybe combining these?
Bincode `serialize_into`
https://docs.rs/bincode/latest/bincode/fn.serialize_into.html
With `write_count`:
https://docs.rs/count-write/latest/count_write/

Reply:
The cost of `serialize_into` is prohibitive.
Just use https://docs.rs/memuse/latest/memuse/